data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -100,9 +107,13 @@ public StreamWorker(final RecordBufferWriter recordBufferWriter,
         this.successItemsCounter = pluginMetrics.counter(SUCCESS_ITEM_COUNTER_NAME);
         this.failureItemsCounter = pluginMetrics.counter(FAILURE_ITEM_COUNTER_NAME);
         this.bytesReceivedSummary = pluginMetrics.summary(BYTES_RECEIVED);
+        this.executorService = Executors.newSingleThreadExecutor(BackgroundThreadFactory.defaultExecutorThreadFactory("mongodb-stream-checkpoint"));
         if (sourceConfig.isAcknowledgmentsEnabled()) {
             // starts acknowledgement monitoring thread
             streamAcknowledgementManager.init((Void) -> stop());
+        } else {
+            // checkpoint in separate thread
+            this.executorService.submit(this::checkpointStream);
```

Comment:
Why are we only adding this thread with acknowledgments enabled? I know checkpointing is done in the callback, but we definitely can't rely on that because if no records are received, we will still just block on `cursor.next()` and not have acknowledgments getting called back to checkpoint, so you will lose ownership

Reply:
When ack is enabled, it's handled in `streamAcknowledgementManager`.

> so you will lose ownership

I currently have this check `!Objects.equals(globalCheckpoint,lastLocalCheckpoint) `. If we remove this check, this checkpoints old value and extends the lease.